Bio/PDB/Entity.py

```diff
@@ -13,20 +13,32 @@
 
 from collections import deque
 from copy import copy
+from typing import TYPE_CHECKING, Any, Dict, Generic, List, Optional, TypeVar, Union
 
 import numpy as np
 
 from Bio import BiopythonWarning
 from Bio.PDB.PDBExceptions import PDBConstructionException
 
+if TYPE_CHECKING:
+    from Bio.PDB.Atom import Atom
 
-class Entity:
+Child = TypeVar("Child", bound=Union["Entity", "Atom"])
+Parent = TypeVar("Parent", bound=Optional["Entity"])
```

Comment:
Maybe make these private for now, rather than expanding the public API? i.e. Add leading underscore.